lib/APPSensorFusion/App.cpp

```diff
@@ -164,6 +166,19 @@ void App::sendSensorData() const
     payload.magnetometerValueY  = magnetometerValues.valueY;
     payload.turnRate            = turnRates.valueZ;
 
+    uint32_t duration = 0U;
+    if (true == m_firstIteration)
+    {
+        duration         = SEND_SENSOR_DATA_PERIOD;

```

Comment:
Instead of doing this, you can use timer.start(NEW_PERIOD), instead or timer.restart(). 
This way you dont need the bool flag.
Example: https://github.com/BlueAndi/DroidControlShip/blob/08cdf8b057bcfd6db7ade3e2961a8fdadce5fa5e/lib/PlatoonService/src/HeadingFinder.cpp#L137